### PR TITLE
Refactor PixelAvatars smart contract and add tests

### PR DESCRIPTION
**Pull request by carlomigueldy**
I've partially written some tests and refactored it but I don't know how I can create a mock when the smart contract calls a method `ownerOf` from DevDaoContract.

image

### CODE REVIEW

**Comment by etherinsight** [comment]
I tested the refactored contract and was successfully able to deploy and mint tokens. LGTM!

**Comment by carlomigueldy** [comment]
Whoah!! Thanks for testing it out @etherinsight !!

**Comment by etherinsight** [comment]
@carlomigueldy Can we uncomment the Rinkeby network on the hardhat.config.ts file? I can also make that change in another PR but if you can do it here that'd be great too. 

```
    // rinkeby: {
    //   url: RINKEYBY_RPC_URL,
    //   accounts: [`0x${RINKEBY_PRIVATE_KEY}`],
    // },


```

**Comment by carlomigueldy** [comment]
Yes sure I will uncomment that @etherinsight

**Comment by carlomigueldy** [comment]
Done @etherinsight !

File: pixel-avatar-hardhat/package.json [hunk 1]
```diff
@@ -4,28 +4,30 @@
...
+    "chain": "hardhat node",
+    "deploy": "hardhat run scripts/deploy.ts --network localhost",
+    "deploy:local": "hardhat run scripts/deploy.ts --network localhost",
+    "deploy:rinkeby": "hardhat run scripts/deploy.ts --network rinkeby",
+    "compile": "hardhat compile",
     "test": "hardhat test"
   },
   "keywords": [],
   "author": "Developer DAO",
+  "private": true,
```
Comment (Dhaiwat10) [inline pixel-avatar-hardhat/package.json:16]: Any reason why this line was added?

**Review by Dhaiwat10 — APPROVED** [review]
LGTM!

File: pixel-avatar-hardhat/package.json (hunk 1 above)
Comment (carlomigueldy) [reply pixel-avatar-hardhat/package.json:16]: Ow, I don't remember @Dhaiwat10 
I can remove that

**Comment by etherinsight** [comment]
Closes Issue #33

**Pull request merged**